packages/front-end/services/scheduleRules.ts

```diff
@@ -0,0 +1,24 @@
+import { FeatureRule, ScheduleRule } from "back-end/types/feature";
+
+export function getUpcomingScheduleRule(
+  rule: FeatureRule
+): ScheduleRule | null {
+  if (!rule.scheduleRules || !rule.scheduleRules.length) {
+    return null;
+  }
+  const currentDate = new Date().valueOf();
+
+  const nextRuleIndex = rule.scheduleRules.findIndex(
+    (rule) =>
```

Comment:
Calling this `rule` is a little confusing since you already have a rule variable that means something else.

Reply:
Updated to `scheduleRule` to be more specific.